Sources/Purchasing/Configuration.swift

```diff
@@ -243,6 +246,30 @@ import Foundation
             return self
         }
 
+        /// Set `validateConfigurationOnLaunch`. Enabled by default.
+        /// If enabled, the SDK will automatically validate its configuration on every launch 
+        /// by running a health check and logging the results to the console. This helps identify
+        /// configuration issues early in the development process.
+        /// 
+        /// The health check validates:
+        /// - API key validity
+        /// - Bundle ID configuration
+        /// - Product configuration in App Store Connect
+        /// - Offering configuration
+        /// - Payment authorization status
+        /// 
+        /// Health check results are logged with appropriate log levels:
+        /// - `.error` for unhealthy configurations that need immediate attention
+        /// - `.warn` for healthy configurations with warnings
+        /// - `.info` for healthy configurations without issues
+        /// 
+        /// - Note: This feature is only available in DEBUG builds
+        /// - Note: Health checks are performed asynchronously and don't block app startup
+        @objc public func with(validateConfigurationOnLaunch: Bool) -> Builder {
+            self.validateConfigurationOnLaunch = validateConfigurationOnLaunch
+            return self
+        }
```

Comment:
Probably we'll want this only `#if !ENABLE_CUSTOM_ENTITLEMENT_COMPUTATION` and make `validateConfigurationOnLaunch = false` by default when `ENABLE_CUSTOM_ENTITLEMENT_COMPUTATION`

Reply:
Very good point, just out of curiosity, is there any docs around the `ENABLE_CUSTOM_ENTITLEMENT_COMPUTATION` and what it is used for? Just so that I can understand it more for future contributions.